src/components/Firestore/CollectionFilter.tsx

```diff
@@ -0,0 +1,455 @@
+    {
+      label: '(==) equal to',
+      value: '==',
+    },
+    {
+      label: '(>) greater than',
+      value: '>',
+    },
+    {
+      label: '(>=) greater than or equal to',
+      value: '>=',
+    },
+    {
+      label: '(<) less than',
+      value: '<',
+    },
+    {
+      label: '(<=) less than or equal to',
+      value: '<=',
+    },
+    {
+      label: '(in) equal to any of the following',
+      value: 'in',
+    },
+    {
+      label: '(array-contains) an array containing',
+      value: 'array-contains',
+    },
+    {
+      label: '(array-contains-any) an array containing any',
+      value: 'array-contains-any',
+    },
+  ];
+
+  return (
+    <>
+      <Controller
+        as={SelectField}
+        name="operator"
+        label="Only show documents where the specified field is..."
+        placeholder="No condition"
+        options={options}
+        onChange={([selected]) => selected.currentTarget.value || undefined}
+      />
+      {children}
+    </>
+  );
+};
+
+const ConditionEntries: React.FC<{ name: string }> = ({ name }) => {
+  const { fields, append, remove } = useFieldArray({
+    name,
+  });
+
+  useEffect(() => {
+    if (fields.length < 1) {
+      append({ _: '' });
+    }
+  }, [append, fields]);
+
+  return (
+    <ul className={styles.conditionEntries}>
+      {fields.map((field, index) => (
+        <li key={field.id}>
+          <ConditionEntry name={`${name}[${index}]`} />
+          <IconButton
+            className={fields.length > 1 ? styles.removeFilter : styles.hidden}
+            icon="delete"
+            label="Remove filter"
+            type="button"
+            onClick={() => remove(index)}
+          />
+        </li>
+      ))}
+      <Button type="button" icon="add" onClick={() => append({ _: '' })}>
+        Add value
+      </Button>
+    </ul>
+  );
+};
+
+const SortRadioGroup: React.FC<{ name: string; disabled: boolean }> = ({
+  name,
+  disabled,
+}) => {
+  const { register, setValue, unregister, watch } = useFormContext();
+
+  useEffect(() => {
+    if (!disabled) {
+      register({ name });
+    } else {
+      setValue(name, undefined);
+    }
+
+    return () => unregister(name);
+  }, [register, unregister, disabled, name, setValue]);
+
+  const sort = watch(name);
+
+  return (
+    <>
+      <Radio
+        value="asc"
+        label="Ascending"
+        checked={sort === 'asc'}
+        onChange={() => setValue(name, 'asc')}
+        disabled={disabled}
+      />
+
+      <Radio
+        value="desc"
+        label="Descending"
+        checked={sort === 'desc'}
+        onChange={() => setValue(name, 'desc')}
+        disabled={disabled}
+      />
+    </>
+  );
+};
+
+function getConditionEntryType(value: any) {
+  if (isBoolean(value)) {
+    return 'boolean';
+  }
+  if (isNumber(value)) {
+    return 'number';
+  }
+  return 'string';
+}
+
+const ConditionEntry: React.FC<{
```

Comment:
nit: Maybe extract the Condition* and Radio group components to new file(s).

This file is getting pretty massive.

Reply:
Yeah I tend to agree with you, but do you think the right ™️  thing to do would be turn CollectionFilter into a folder with Condition* and Radio components? I'm not sold on adding those to the top-level Firestore/ directory.

Reply:
Yep, definitely +1 to grouping under a folder.

Reply:
Cool, broke it up into smaller bits under a new CollectionFilter/ directory